completions/cryptsetup

```diff
@@ -11,6 +11,50 @@ _cryptsetup_device()
     _filedir
 }
 
+_cryptsetup_action()
+{
+    local ret IFS=$' \t\n'
+    _comp_dequote "${1-}" || return 1
+    local cmd=${ret:-cryptsetup}
+    local actions=$(
+        {
+            LC_ALL=C "$cmd" --help 2>&1 |
+                sed -n '/^<action> is one of:/,/^[^[:space:]]/s/^[[:space:]]\{1,\}\([^[:space:]]\{1,\}\).*/\1/p'
+            LC_ALL=C man cryptsetup 2>&1 |
+                awk '/^[[:space:]]+[[:alnum:]_]+([[:space:]]+(-[^[:space:].]+|<[^<>]+>|\[[^][]+\]|or))*$/ {print $1}'
+        } | sort -u
+    )
+
+    if [[ ! $actions ]]; then
+        # The fallback action list is extracted from the following source:
+        # https://gitlab.com/cryptsetup/cryptsetup/-/blob/main/src/cryptsetup.c#L3154-3208 (search for "Handle aliases")
+        # https://gitlab.com/cryptsetup/cryptsetup/-/blob/main/src/cryptsetup.c#L2831-2867 (search for "struct action_type")
+        # https://gitlab.com/cryptsetup/cryptsetup/-/blob/main/src/cryptsetup_args.h#L28-53 (see the macros "*_ACTION")
+        actions='benchmark bitlkClose bitlkDump bitlkOpen close config convert
+            create erase isLuks loopaesClose loopaesOpen luksAddKey
+            luksChangeKey luksClose luksConfig luksConvertKey luksDump
+            luksErase luksFormat luksHeaderBackup luksHeaderRestore
+            luksKillSlot luksOpen luksRemoveKey luksResume luksSuspend luksUUID
+            open plainClose plainOpen reencrypt refresh remove repair resize
+            status tcryptClose tcryptDump tcryptOpen token'
+
+        # We attempt to filter the supported actions by the strings in the binary.
+        local path
+        if path=$(type -P -- "$cmd" 2>/dev/null || type -P -- cryptsetup 2>/dev/null); then
+            local filtering_pattern
+            printf -v filtering_pattern '%s\n' $actions
+            filtering_pattern=${filtering_pattern%$'\n'}
+
+            local filtered_actions
+            filtered_actions=$(strings "$path" | command grep -Fx "$filtering_pattern" | sort -u) &&
```

Comment:
If we decide to keep this block or to extract this as a helper to another PR, I suppose this would be in order:
```suggestion
            filtered_actions=$(strings "$path" 2>/dev/null | command grep -Fx "$filtering_pattern" | sort -u) &&
```

Reply:
I'm not sure if the fallback in this function finally gets into the repository, but such a function would be useful anyway where hardcoded lists are needed. I'll later consider it.